tracer/test/test-applications/integrations/Samples.Kafka/Program.cs

```diff
@@ -122,6 +125,53 @@ await Task.WhenAny(
                 Task.Delay(TimeSpan.FromSeconds(5)));
         }
 
+        private static async Task TryProduceWithNullTopicPartition(string topic, ClientConfig config)
+        {
+            // NOTE: you'll see an Null Ref exception, it is expected, but it shouldn't be in OUR logs
+            try
+            {
+                using var producer = new ProducerBuilder<string, string>(config).Build();
+
+                // Create a message with null values but valid structure
+                var message = new Message<string, string>
+                {
+                    Key = null,
+                    Value = null,
+                };
+
+                // This should pass Kafka validation but might still trigger our issue
+                Console.WriteLine("Attempting to produce message with NULL TopicPartition THIS SHOULD THROW.");
+                await producer.ProduceAsync((TopicPartition)null, message);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("CAUGHT EXPECTED EXCEPTION!");
+                Console.WriteLine($"Exception in reproduction test: {ex}");
+                Console.WriteLine($"Stack trace: {ex.StackTrace}");
```

Comment:
nit: this already prints the stack
```suggestion
                Console.WriteLine($"Exception in reproduction test: {ex}");
```